### PR TITLE
Keep Track Anything tensors on GPU

### PR DESCRIPTION
**Pull request by nkumarrai-nv**
This PR removes avoidable CPU/GPU round trips from the Track Anything pipeline by adding CUDA tensor-native paths through detection, segmentation, and tracking.

  Key changes:

  - Keep VideoFrame.rgb, detection boxes, SAM masks, AOT masks, and tracker outputs on GPU.
  - Add GPU-direct GroundingDINO preprocessing/inference instead of PIL/NumPy transforms in the hot path.
  - Add CUDA tensor input support for SAM image embedding and batched box-prompt segmentation.
  - Replace AOT’s CPU transform path with tensor-native resize/normalization and CUDA mask handling.
  - Update object-mask merging/new-object detection to operate on torch tensors.
  - Add small shared-memory padding in altcorr_index_forward_kernel.

  Behavioral note:
  - The Track Anything path now expects CUDA RGB tensors and fails fast if called with CPU/NumPy inputs.

### CODE REVIEW

File: csrc/droid_net_ext/altcorr_kernel.cu [hunk 1]
```diff
@@ -161,8 +161,8 @@ __global__ void altcorr_index_forward_kernel(
     const int W2 = fmap2.size(3);
     const int C = fmap1.size(4);
 
-    __shared__ scalar_t f1[CHANNEL_STRIDE][BLOCK_HW];
-    __shared__ scalar_t f2[CHANNEL_STRIDE][BLOCK_HW];
+    __shared__ scalar_t f1[CHANNEL_STRIDE][BLOCK_HW + 1];
+    __shared__ scalar_t f2[CHANNEL_STRIDE][BLOCK_HW + 1];
```
Comment (heiwang1997) [inline csrc/droid_net_ext/altcorr_kernel.cu:171]: Can you elaborate why is this `+1` needed?

File: vipe/priors/track_anything/aot_tracker.py [hunk 2]
```diff
@@ -46,34 +44,80 @@ def _build_aot_model():
...
+        self.max_short_edge = cfg.TEST_MAX_SHORT_EDGE
+        self.max_long_edge = cfg.TEST_MAX_LONG_EDGE
+        self.flip = cfg.TEST_FLIP
+        self.multi_scale = cfg.TEST_MULTISCALE
+        self.align_corners = cfg.MODEL_ALIGN_CORNERS
+        self.max_stride = 16
 
         self.model.eval()
 
+    def _restricted_size(self, height: int, width: int) -> tuple[int, int]:
```
Comment (heiwang1997) [inline vipe/priors/track_anything/aot_tracker.py:56]: Why are we re-implementing the features in `from .aot.transforms import video_transforms as tr` here? Is it because the original ones are `numpy`-based?

**Review by heiwang1997 — COMMENTED** [review]
Looks good! For sanity, is it possible for us to remove all the dead `numpy` code paths, and force `torch.Tensor` to be passed around consistently?

File: csrc/droid_net_ext/altcorr_kernel.cu (hunk 1 above)
Comment (nkumarrai-nv) [reply csrc/droid_net_ext/altcorr_kernel.cu:171]: This was a suggestion from codex.

Here is the reason - 

The BLOCK_HW + 1 change is shared-memory padding, not a correctness fix. The kernel only indexes columns 0..BLOCK_HW-1; the extra column is unused data-wise.

  It helps avoid shared-memory bank conflicts in accesses like:

  f1[tid % CHANNEL_STRIDE][k1]

With BLOCK_HW = 32, each row has stride 32. When a warp accesses the same column across 32 rows, all lanes map to the same shared-memory bank, causing a 32-way conflict. Padding the row to 33 shifts each row by one bank, so those same-column row accesses spread across banks and can be served in parallel.

This mainly benefits the shared-memory tile load/store pattern in altcorr_index_forward_kernel. The dot-product reads f1[k][tid] are a different access pattern and are not the main reason for the padding.

File: vipe/priors/track_anything/aot_tracker.py (hunk 2 above)
Comment (nkumarrai-nv) [reply vipe/priors/track_anything/aot_tracker.py:56]: Yes, all the transforms had numpy / cv based ops.

File: csrc/droid_net_ext/altcorr_kernel.cu (hunk 1 above)
Comment (heiwang1997) [reply csrc/droid_net_ext/altcorr_kernel.cu:171]: ok that makes sense. Let's make sure that this is documented.

Comment (nkumarrai-nv) [reply csrc/droid_net_ext/altcorr_kernel.cu:171]: Done
